functional-tests/src/test/java/org/infinispan/online/service/caching/CachingServiceTest.java

```diff
@@ -29,30 +32,60 @@
    @ArquillianResource
    URL restService;
 
+   @Named("caching-service-app-memcached")
+   @ArquillianResource
+   URL memcachedService;
+
    @ArquillianResource
    OpenShiftClient openShiftClient;
 
    ReadinessCheck readinessCheck = new ReadinessCheck();
    HotRodTester hotRodTester = new HotRodTester();
    RESTTester restTester = new RESTTester();
+   MemcachedTester memcachedTester = new MemcachedTester();
 
    @Before
    public void before() {
       readinessCheck.waitUntilAllPodsAreReady(openShiftClient, 30, TimeUnit.SECONDS);
    }
 
    @Test
-   public void should_default_cache_be_accessible_via_hot_rod() throws IOException {
-      hotRodTester.testBasicEndpointCapabilities(hotRodService);
+   @Ignore
+   public void should_not_blow_up_because_of_oom() {
+      hotRodTester.testPutPerformance(hotRodService, 60, TimeUnit.SECONDS);
+   }
+
+   @Test
+   public void restBasicTypesPutGet() throws IOException {
+      restTester.putGetTest(restService);
    }
 
    @Test
-   public void should_not_blow_up_because_of_oom() {
-      hotRodTester.testPutPerformance(hotRodService, 60, TimeUnit.SECONDS);
+   public void hotRodPutGet() {
+      hotRodTester.putGetTest(hotRodService);
    }
 
    @Test
-   public void should_default_cache_be_accessible_via_REST() throws IOException {
-      restTester.testBasicEndpointCapabilities(restService);
+   public void memcachedPutGetTest() {
+      memcachedTester.putGetTest(memcachedService);
+   }
+
+   // The eviction can not be turned off on caching service
+   // The test puts entries to the cache until one is evicted
+   @Ignore
+   @Test(timeout = 600000)
+   public void evictionTest() {
```

Comment:
I've been thinking about this test and it needs to work differently...

At first you need inspect what is the `request` limit set in the Pod (you may use `OpenShiftClient`, find proper Pod and check its Containers specification). The next step is to calculate how much data you need to put to test eviction. Here's an example - Memory request is set to 512 MB, typical JVM overhead is ~200-300 MB (see https://github.com/jboss-container-images/jboss-dataservices-image/blob/dataservices-dev/modules/os-datagrid-adjust-memory/added/adjust_memory.sh). This means that you need to insert at least 300MB of data to test it properly. 

Of course, you also need a time limit in case when this test hangs. But set it to some very high number.

Reply:
@slaskawi   I'm not sure I see the added value. Right now, the test runs the shortest possible time - until the first entry is evicted.
It seems like a lot of overhead to calculate/estimate how much data is needed, put it there and then check the same thing we are checking now. Plus if some memory options would change (other than container memory limit) it would break the test.
I would prefer to keep it this way, but if you have a good reason, I will change it.

Reply:
@slaskawi @rmacor it looks like it is too early to write this test. There's now the email thread about eviction and seems the current settings are not reliable. First it needs some tuning. I also think that the ultimate goal of this test is to prove that the service will not go out-of-memory. Loading data for a sufficiently long time seems good enough. I'd just not rely on the fact that evicting the first entry means that the container will not go OOM. It merely proves that eviction is turned on. That might be the reason why on one machine the test passes and on another machine it fails. The test needs to load more data and keep evicting for some time.

Reply:
Hmm, apparently there's already a test for OOM so this test could be just modified to check that some entries have really been evicted.

Reply:
Yes, I'm still working on https://issues.jboss.org/browse/ISPN-8397 to stabilize eviction based on memory parameters. I hope I will finish this work till tomorrow.

Reply:
@rmacor hmmm until first entry gets evicted? How do you check this?

Reply:
@slaskawi I remember the first entry once is no longer in the cache, it's been evicted.
while (byteArrayCache.get(key) != null) {

Reply:
@rmacor Nice trick! +1